internal/sqld/sqldwebsockets/websockets.go

```diff
@@ -0,0 +1,230 @@
+package sqldwebsockets
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+
+	"nhooyr.io/websocket"
+	"nhooyr.io/websocket/wsjson"
+	"vitess.io/vitess/go/pools"
+)
+
+func errorMsg(errorResp interface{}) string {
+	return errorResp.(map[string]interface{})["error"].(map[string]interface{})["message"].(string)
+}
+
+func isErrorResp(resp interface{}) bool {
+	return resp.(map[string]interface{})["type"] == "response_error"
+}
+
+type SqldWebsocket struct {
+	conn   *websocket.Conn
+	idPool *pools.IDPool
+}
+
+type Params struct {
+	Names  []string
+	Values []any
+}
+
+func convertValue(v any) (map[string]interface{}, error) {
+	res := map[string]interface{}{}
+	if v == nil {
+		res["type"] = "null"
+	} else if integer, ok := v.(int64); ok {
+		res["type"] = "integer"
+		res["value"] = strconv.FormatInt(integer, 10)
+	} else if text, ok := v.(string); ok {
+		res["type"] = "text"
+		res["value"] = text
+	} else if blob, ok := v.([]byte); ok {
+		res["type"] = "blob"
+		res["value"] = base64.StdEncoding.EncodeToString(blob)
+	} else if float, ok := v.(float64); ok {
+		res["type"] = "float"
+		res["value"] = float
+	} else {
+		return nil, fmt.Errorf("unsupported value type: %s", v)
+	}
+	return res, nil
+}
+
+type ExecResponse struct {
+	resp map[string]interface{}
+}
```

Comment:
Instead of treating the response as an untyped `map[string]interface{}`, it should be possible to model the response as a Go struct an use the built-in serde-like functionality to deserialize directly into this struct, instead of manual parsing of the JSONs.
https://pkg.go.dev/encoding/json

Reply:
I thought it might me more efficient to just look into particular fields that are used instead of parsing the whole message straight away.

Reply:
Also, there should be few cases where you can ignore some fields in the exchanged messages.